ssv/spectest/generate/main.go

```diff
@@ -72,12 +73,23 @@ func clearStateComparisonFolder() {
 }
 
 func writeJsonStateComparison(name, testType string, post interface{}) {
+	postMap, ok := post.(map[string]types.Root)
```

Comment:
why not pass it as `map[string]types.Root` to the function then ?

Reply:
ideally I'd split it to two functions and check before calling. go is meant to be statically typed. Generics introduced not long ago, I think we should either stick to using types or used generics.

Reply:
This is more problematic than it first seems.
`post` is received from `GetPostState()` that returns an `interface{}`

`GetPostState()` is an interface function that sometimes returns a `map[string]types.Root`, and sometimes returns `types.Root` (it may also return a slice for now, but it should be deprecated).

Since the point is to not use `interface{}` at all I think to change `getPostState` to always return a `map`. Where we originally returned `Root` it will simply be a map size of 1.

What do you think?

Reply:
Opened #283 to track refactoring changed.
I want to solve each issue in ints own PR to cause less disruptions and ensure quality

Reply:
you could also return some interface or struct that could have both of them. staying with only map sounds ok aswell.